### PR TITLE
[Arm64] Add GT_MULHI

### LINKED ISSUE #13820: [Arm64] Implement genCodeForMulHi and enable magic division
`NYI("genCodeForMulHi");`

@dotnet/arm64-contrib @dotnet/jit-contrib Please assign to me.

### PR DESCRIPTION
**Pull request by sdmaclea**
Fix ` NYI("genCodeForMulHi")` and enable related lowerings

Fixes #13820

### CODE REVIEW

**Comment by sdmaclea** [comment]
@dotnet/arm64-contrib @dotnet/jit-contrib PTAL

**Comment by sdmaclea** [comment]
@dotnet-bot test this please

**Comment by sdmaclea** [comment]
@dotnet-bot test Tizen armel Cross Debug Build

**Comment by sdmaclea** [comment]
@briansull PTAL

**Comment by jashook** [comment]
@donet-bot test Windows_NT Arm64 Checked

**Comment by jashook** [comment]
@donet-bot test Windows_NT Arm64 Checked

File: src/jit/codegenarm64.cpp [hunk 1]
```diff
@@ -1502,53 +1502,44 @@ void CodeGen::genSetRegToConst(regNumber targetReg, var_types targetType, GenTre
...
-    {
-        regOp = op2;
-        rmOp  = op1;
-    }
-    assert(!regOp->isContained());
-            
-    // Setup targetReg when neither of the source operands was a matching register
-    if (regOp->gtRegNum != targetReg)
+    // The arithmetic node must be sitting in a register (since it's not contained)
+    noway_assert(targetReg != REG_NA);
```
Comment (CarolEidt) [inline src/jit/codegenarm64.cpp]: Does this need to be a noway_assert?

File: src/jit/codegenarm64.cpp [hunk 2]
```diff
@@ -1502,53 +1502,44 @@ void CodeGen::genSetRegToConst(regNumber targetReg, var_types targetType, GenTre
...
+
+    if (EA_SIZE(attr) == EA_8BYTE)
     {
-        inst_RV_RV(ins_Copy(targetType), targetReg, regOp->gtRegNum, targetType);
+        instruction ins = isUnsigned ? INS_umulh : INS_smulh;
+
+        regNumber r = emit->emitInsTernary(ins, attr, treeNode, op1, op2);
+
+        noway_assert(r == targetReg);
```
Comment (CarolEidt) [inline src/jit/codegenarm64.cpp]: Ditto - this should be a given for `emitInsTernary()`, I would think.

File: src/jit/codegenarm64.cpp (hunk 1 above)
Comment (sdmaclea) [reply src/jit/codegenarm64.cpp]: @CarolEidt Those were copied from genCodeForBinary, I will remove all four.

Comment (sdmaclea) [reply src/jit/codegenarm64.cpp]: Done

**Comment by sdmaclea** [comment]
@donet-bot test Windows_NT Arm64 Checked

File: src/jit/codegenarm64.cpp (hunk 1 above)
Comment (CarolEidt) [reply src/jit/codegenarm64.cpp]: Thanks!

**Comment by sdmaclea** [comment]
@donet-bot test Windows_NT Arm64 Checked

**Comment by sdmaclea** [comment]
All tests have passed including Windows_NT Arm64 Checked.  This can be merged.

**Pull request merged**